server/src/main/java/org/elasticsearch/action/datastreams/lifecycle/ExplainIndexDataStreamLifecycle.java

```diff
@@ -151,6 +160,9 @@ public XContentBuilder toXContent(
     public void writeTo(StreamOutput out) throws IOException {
         out.writeString(index);
         out.writeBoolean(managedByLifecycle);
+        if (out.getTransportVersion().onOrAfter(TransportVersions.NO_LIFECYCLE_FOR_SYSTEM_DATA_STREAMS)) {
+            out.writeBoolean(isSystemDataStream);
+        }
         if (managedByLifecycle) {
             out.writeOptionalLong(indexCreationDate);
```

Comment:
I would have assumed the `isSystemDataStream` serialization would have had to come at the end, but because the other fields are conditionally written, I guess the serialization needs to come before those optional fields?

Reply:
I don't think it makes any difference. I put it there because everything else in the serialization was conditional on the existence of `managedByLifecycle`, but that was just personal preference.